src/cache/asset.rb

```diff
@@ -25,8 +25,16 @@ def asset_for_node(node)
         data[:node][node.name.to_sym]
       end
 
-      def unassign_asset(asset_name)
-        data.delete_if { |_key, value| value == asset_name }
+      def unassign_asset(asset_name, *node_name)
+        data.each_value do |value|
+          value.delete_if do |node, asset|
+            if node_name.empty?
+              asset == asset_name
+            else
+              node == node_name[0].to_sym && asset == asset_name
```

Comment:
This line is very dense, and also duplicates an equality check. To fix this, the equality checks should be extracted.

As we are only interested in assets of a particular name, the block can start with a guard statement:
` next unless asset == asset_name`

Assuming my above comment has been addressed, than the return value can be obtained with:
`node_name ? (node == node_name) : true`